### ISSUE TITLE
Futhark type parameters seem to be lexed as character literals

### ISSUE THREAD

**Issue by athas** [Alhadis/language-etc#6]
Type parameters in Futhark are written as `'x` [as seen on this line](https://github.com/diku-dk/futhark/blob/bdd7215ca3df28c857807feb4b9ed94e3f278e75/tests/types/typeparams0.fut#L5).  Character literals (which are extremely rare in Futhark) are written as `'x'`, so I think the lexer gets confused.  I'm not quite sure where this goes wrong though, since [the character rule](https://github.com/Alhadis/language-etc/blob/05d8ca78a6f5a5d90c415d536b541603d9b93737/grammars/futhark.cson#L106) looks like it correctly only detect character literals that are a single character enclosed in single quotes.

**Comment by Alhadis** [comment]
Fixed in 2806cb6. :+1: Thanks for bringing this to my attention!

> I'm not quite sure where this goes wrong though

[This line](https://github.com/Alhadis/language-etc/blob/05d8ca78a6f5a5d90c415d536b541603d9b93737/grammars/futhark.cson#L42) is importing [these patterns](https://github.com/Alhadis/language-etc/blob/05d8ca78a6f5a5d90c415d536b541603d9b93737/grammars/etc.cson#L3-L13), one of which is a rule for matching single-quoted strings:

https://github.com/Alhadis/language-etc/blob/05d8ca78a6f5a5d90c415d536b541603d9b93737/grammars/etc.cson#L455-L462

I added `include: "etc"` to "mop up" anything that wasn't matched by the Futhark grammar (mainly punctuation characters). At the time, I wasn't aware that type parameters existed (or they may not have, assuming they were introduced to the language after the creation of this grammar). I suspect it's the latter, since the [language reference](https://github.com/diku-dk/futhark/blob/bdd7215ca3df28c857807feb4b9ed94e3f278e75/docs/language-reference.rst) explains type parameters in a hard-to-miss fashion. Curiously, it also omits any mention of the character literal syntax.
